### DISCUSSION TITLE
String.IndexOf() and NewLine '\n'

### DISCUSSION THREAD

**Discussion by robloo** [dotnet/runtime#49586] [Q&A] [answered]
Really basic yet interesting question for me.

Why does the following code return -1 when running on EN-US culture? I would expect it to find the newline character and return index 1.

```"\r\n".IndexOf("\n");```

Of course this is bad practice and we should be specifying StringComparison.Ordinal or StringComparison.OrdinalIgnoreCase. However, I genuinely don't understand why this doesn't work without specifying the StringComparison. 

Note that the following work just fine:

```
"\r\n".IndexOf("\r\n");
"\r\n".IndexOf(Environment.NewLine);
```

**Comment by stephentoub** [comment, marked as answer]
See https://github.com/dotnet/runtime/issues/43736

**Reply by robloo** [reply]
Thanks, that explains the issue and why it wasn't noticed earlier.

This was found after moving some tests to .NET 5. I'm glad the main codebase hasn't been ported yet. I wouldn't expect such a breaking change to occur especially one that doesn't seem to make much sense.
